packages/@aws-cdk/aws-dynamodb/lib/replica-handler/index.ts

```diff
@@ -5,27 +5,37 @@ import { DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-
 export async function onEventHandler(event: OnEventRequest): Promise<OnEventResponse> {
   console.log('Event: %j', event);
 
-  /**
-   * Process only Create and Delete requests. We shouldn't receive any
-   * update request and in case we do there is nothing to update.
-   */
+  const dynamodb = new DynamoDB();
+
+  let updateTableAction: 'Create' | 'Update' | 'Delete';
+
   if (event.RequestType === 'Create' || event.RequestType === 'Delete') {
-    const dynamodb = new DynamoDB();
-
-    const data = await dynamodb.updateTable({
-      TableName: event.ResourceProperties.TableName,
-      ReplicaUpdates: [
-        {
-          [event.RequestType]: {
-            RegionName: event.ResourceProperties.Region,
-          },
+    updateTableAction = event.RequestType;
+  } else { // Update
+    // This can only be a table replacement so we create a replica
+    // in the new table. The replica for the "old" table will be
+    // deleted when CF issues a Delete event on the old physical
+    // resource id.
+    updateTableAction = 'Create';
+  }
+
+  const data = await dynamodb.updateTable({
+    TableName: event.ResourceProperties.TableName,
+    ReplicaUpdates: [
+      {
+        [updateTableAction]: {
+          RegionName: event.ResourceProperties.Region,
         },
-      ],
-    }).promise();
-    console.log('Update table: %j', data);
+      },
+    ],
+  }).promise();
+  console.log('Update table: %j', data);
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    return { PhysicalResourceId: `${event.ResourceProperties.TableName}-${event.ResourceProperties.Region}` };
```

Comment:
This is actually 2 functional changes, correct?

1. Previously, we returned `PhysicalResourceId` also for deletes, now we don't.
2. We used to return `event.ResourceProperties.Region` as the physical ID, now we return `` `${event.ResourceProperties.TableName}-${event.ResourceProperties.Region}` ``.

So, is 1. intentional, or accidental? And what about 2. - what will be the behavior for existing customers of global DB Tables?

Reply:
For existing users of global DB tables if they update to new the code the only thing that will happen is the replacement of the inline policies with managed policies. This is because nothing changes on the properties of the custom resource so it won't receive any event (Create/Update/Delete).

Now if later a user decides to remove one or more replica regions, the custom resource will receive a Delete event. If we don't change how the physical resource id is returned for a Delete event then CF will receive a new physical resource id (`table-region` instead of `region`). Changing the physical resource id in a Delete event is not allowed by CF and the Delete will fail (from a CF perspective and this also means that it will not wait for `isComplete`). So yes 1. is intentional.

Returning an empty object on delete ensures that the custom resource provider framework uses the physical resource id from the event (should have been done in the first place).